docs/config_entries_config_flow_handler.md

```diff
@@ -325,14 +325,18 @@ class OAuth2FlowHandler(
 
     reauth_entry: ConfigEntry | None = None
 
-    async def async_step_reauth(self, user_input=None):
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
```

Comment:
Add type hints, and highlight that the initial input contains the data from the config entry.